test/test_misc_utils.py

```diff
@@ -1282,6 +1288,16 @@ def test_check_true():
     assert msg in str(e)
 
 
+def test_remove_element():
+
+    assert remove_element('b', ['a', 'b', 'c', 'a', 'b', 'c']) == ['a', 'c', 'a', 'b', 'c']
+
+    with pytest.raises(ValueError):
+        assert remove_element('z', ['a', 'b', 'c', 'a', 'b', 'c'])
+
+    assert remove_element('z', ['a', 'b', 'c', 'a', 'b', 'c'], raise_error=False) == ['a', 'b', 'c', 'a', 'b', 'c']
+
```

Comment:
I would test the "copy" part of this ie:
```
lst = [1, 2, 3]
new_lst = remove_element(1, lst)
assert lst == [1, 2, 3]
assert new_lst == [2, 3]
```

Reply:
Might be worth having a version that removes from the source list as well (no copy).

Reply:
You're right I should test the original list.

Reply:
I had specific need of the non-side-effecting one.  I'm less certain about the side-effecting one, so I'm going to leave that until we have a need so I can be sure I understand the situation. Here's the test I'm now doing:
```
    old = ['a', 'b', 'c', 'a', 'b', 'c']
    new = remove_element('b', old)
    assert old is not new
    assert old == ['a', 'b', 'c', 'a', 'b', 'c']
    assert new == ['a', 'c', 'a', 'b', 'c']

    new = remove_element('z', old, raise_error=False)
    assert old is not new
    assert new == ['a', 'b', 'c', 'a', 'b', 'c']
    assert old == ['a', 'b', 'c', 'a', 'b', 'c']

    with pytest.raises(ValueError):
        remove_element('z', old)
    assert old == ['a', 'b', 'c', 'a', 'b', 'c']
```